core/src/main/java/org/hibernate/ogm/loader/nativeloader/impl/BackendCustomQuery.java

```diff
@@ -60,24 +60,27 @@ public BackendCustomQuery(String queryString, T query, NativeSQLQueryReturn[] qu
 			this.querySpaces = Collections.emptySet();
 		}
 
-		this.singleEntityKeyMetadata = determineSingleEntityKeyMetadata( factory, customQueryReturns );
+		this.singleEntityMetadataInformation = determineSingleEntityMetadataInformation( factory, customQueryReturns );
 	}
 
-	private static EntityKeyMetadata determineSingleEntityKeyMetadata(SessionFactoryImplementor sessionFactory, List<Return> customQueryReturns) {
-		EntityKeyMetadata metadata = null;
+	private EntityMetadataInformation determineSingleEntityMetadataInformation(SessionFactoryImplementor sessionFactory, List<Return> customQueryReturns) {
+		EntityMetadataInformation metadataInformation = null;
 
 		for ( Return queryReturn : customQueryReturns ) {
 			if ( queryReturn instanceof RootReturn ) {
-				if ( metadata != null ) {
+				if ( metadataInformation != null ) {
 					return null;
 				}
 				RootReturn rootReturn = (RootReturn) queryReturn;
 				OgmEntityPersister persister = (OgmEntityPersister) sessionFactory.getEntityPersister( rootReturn.getEntityName() );
-				metadata = new DefaultEntityKeyMetadata( persister.getTableName(), persister.getRootTableIdentifierColumnNames() );
+				metadataInformation = new EntityMetadataInformation(
+						new DefaultEntityKeyMetadata( persister.getTableName(), persister.getRootTableIdentifierColumnNames() ),
```

Comment:
Yes, haven't looked after it as it was already instantiating a new one in the existing code.